trace-utils/src/send_data.rs

```diff
@@ -0,0 +1,229 @@
+// Copyright 2024-Present Datadog, Inc. https://www.datadoghq.com/
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::{anyhow, Context};
+use futures::stream::FuturesUnordered;
+use futures::StreamExt;
+use hyper::{Body, Client, Method, Response, StatusCode};
+use std::collections::HashMap;
+
+use crate::tracer_header_tags::TracerHeaderTags;
+use datadog_trace_protobuf::pb;
+use ddcommon::{connector, Endpoint, HttpRequestBuilder};
+
+#[derive(Debug)]
+pub enum SendRequestError {
+    Hyper(hyper::Error),
+    Any(anyhow::Error),
+}
+
+pub struct SendDataResult {
+    pub last_result: anyhow::Result<Response<Body>>,
+    pub requests_count: u64,
+    pub responses_count_per_code: HashMap<u16, u64>,
+    pub errors_timeout: u64,
+    pub errors_network: u64,
+    pub errors_status_code: u64,
+}
+
+impl SendDataResult {
+    fn new() -> SendDataResult {
```

Comment:
Since you're not taking in any parameters, should this just be a `Default` impl?

Reply:
Agreed